### PR TITLE
mtl/ofi: Add missing cq_data_size value in hints

### LINKED ISSUE #8305: OFI MTL: ring_c and connectivity_c hang when using PSM2 provider
In doing some sanity checks of the OFI support in various releases I noticed that the PSM2 provider no longer works on the master and v4.1.x branches.

```
hpp@sn196:/users/hpp/ompi/examples_sn>mpirun --mca mtl_ofi_provider_include sockets -np 4 ./ring_c
Process 0 sending 10 to 1, tag 201 (4 processes in ring)
Process 0 sent to 1
Process 0 decremented value: 9
Process 0 decremented value: 8
Process 0 decremented value: 7
Process 0 decremented value: 6
Process 0 decremented value: 5
Process 0 decremented value: 4
Process 0 decremented value: 3
Process 0 decremented value: 2
Process 0 decremented value: 1
Process 0 decremented value: 0
Process 0 exiting
Process 1 exiting
Process 2 exiting
Process 3 exiting

hpp@sn196:/usr/projects/artab/users/hpp/ompi/examples_sn>mpirun --mca mtl_ofi_provider_include psm2 -np 4 ./ring_c
Process 0 sending 10 to 1, tag 201 (4 processes in ring)
Process 0 sent to 1
```

master (and probably 4.1.x) does work with the GNI and verbs:rxm providers.
v4.0.x works nominally with the psm2 provider

### PR DESCRIPTION
**Pull request by acgoldma**
Fixes #8305
Signed-off-by: Goldman, Adam <adam.goldman@intel.com>

### CODE REVIEW

**Comment by jsquyres** [comment]
ok to test

File: ompi/mca/mtl/ofi/mtl_ofi_component.c [hunk 1]
```diff
@@ -626,6 +626,9 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
         hints->domain_attr->threading = FI_THREAD_DOMAIN;
     }
 
+    if ((MTL_OFI_TAG_AUTO == ofi_tag_mode) || (MTL_OFI_TAG_FULL == ofi_tag_mode))
```
Comment (jsquyres) [inline ompi/mca/mtl/ofi/mtl_ofi_component.c:629]: Style nit: can you put `{}` around the block?  It's Open MPI style that we put `{}` around _all_ blocks, even 1-line blocks.

Thanks!

Comment (acgoldma) [reply ompi/mca/mtl/ofi/mtl_ofi_component.c]: Done, pushed update.

**Comment by hppritcha** [comment]
testing this on our omnipath cluster

**Comment by hppritcha** [comment]
works for me

**Review by hppritcha — APPROVED** [review]

**Review by rajachan — COMMENTED** [review]
This change itself looks good to me. Did you see my comment in https://github.com/open-mpi/ompi/issues/8305#issuecomment-754843648? Even without the hint, the test should not hang.

**Review by rajachan — APPROVED** [review]

**Comment by acgoldma** [comment]
> This change itself looks good to me. Did you see my comment in [#8305 (comment)](https://github.com/open-mpi/ompi/issues/8305#issuecomment-754843648)? Even without the hint, the test should not hang.

Yes, I plan to investigate that as well. This patch in ompi was to preserve similar functionality.

**Review by mwheinz — APPROVED** [review]
Works for me.

**Comment by rajachan** [comment]
This would also have to be cherry-picked to the 4.1.x branch.

**Pull request merged**